pkg/profiler/profiler.go

```diff
@@ -550,15 +511,68 @@ func (p *CgroupProfiler) profileLoop(ctx context.Context, captureTime time.Time)
 		return fmt.Errorf("failed iterator: %w", it.Err())
 	}
 
+	prof, err := p.buildProfile(ctx, captureTime, samples, locations, kernelLocations, userLocations, mappings, kernelMapping)
+	if err != nil {
+		return fmt.Errorf("failed to build profile: %w", err)
+	}
+
+	if err := p.writeProfile(ctx, prof); err != nil {
+		level.Error(p.logger).Log("msg", "failed to send profile", "err", err)
+	}
+
+	if err := p.bpfMaps.clean(); err != nil {
+		level.Warn(p.logger).Log("msg", "failed to clean BPF maps", "err", err)
+	}
+
+	return nil
+}
+
+func (p *CgroupProfiler) loopReport(lastProfileTakenAt time.Time, lastError error) {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	p.lastProfileTakenAt = lastProfileTakenAt
+	p.lastError = lastError
+}
+
+func (p *CgroupProfiler) buildProfile(
+	ctx context.Context,
+	captureTime time.Time,
+	samples map[[254]uint64]*profile.Sample,
```

Comment:
ooooooo we can't use the constant for this 254 here, can we?

Reply:
No, we can't. IDE generated signature, I've overlooked it. I will introduce a specific type for it.